src/commands/synthetics/__tests__/cli.test.ts

```diff
@@ -99,6 +127,28 @@ describe('run-test', () => {
       })
     })
 
+    test('partial retryConfig override from ENV retains existing values', async () => {
+      const overrideEnv = {
+        DATADOG_SYNTHETICS_OVERRIDE_RETRY_COUNT: '5',
+      }
+      process.env = overrideEnv
+      const command = createCommand(RunTestsCommand)
+
+      command['config'].defaultTestOverrides = {
+        ...command['config'].defaultTestOverrides,
+        retry: {
+          count: 1,
+          interval: 42,
+        },
+      }
+      await command['resolveConfig']()
+
+      expect(command['config'].defaultTestOverrides.retry).toEqual({
+        count: 5,
+        interval: 42,
+      })
+    })
```

Comment:
(nit) Technically to have complete test coverage you should have a test case for `interval` as well :D and it would be easy to parameterise the test

However considering you're testing the code path, i can let is slide :D